File: Java/Stack.java
class Node {
    Object val;
    Node next;
    public Node(Object val) {
        this.val = val;
    }
    public String toString() {
        return this.val + "->" + this.next;
    }
}

class Stack < T > {

    Node list;
    Node tail;
    int len = 0;

    public boolean push(T val) {
        Node node = new Node(val);
        if (tail == null) {
            this.list = node;
            this.tail = node;
            this.len++;
            return true;
        }
        tail.next = node;
        tail = node;
        this.len++;
        return true;
    }

    public boolean pop() {
        Node temp = this.list;
        if (len <= 0) return false;
        if (len == 1) {
            this.list = null;
            this.tail = null;
            this.len--;
            return true;
        }
        while (temp.next.next != null) {
            temp = temp.next;
        }
        this.tail = temp;
        this.len--;
        temp.next = null;
        return true;
    }

    public T peek() {
        if (len <= 0) return null;
        return (T)this.tail.val;
    }

    public int size() {
        return this.len;
    }

    public String toString() {
        return this.list + "";
    }

}

File: Javascript/Stack.js
class Node {
    val;
    next;
    constructor(val) {
        this.val = val;
        this.next = null;
    }
}

class Stack {
    head;
    tail;
    len = 0;
    push(val) {
        let node = new Node(val);
        if (this.len === 0) {
            this.len++;
            this.head = node;
            this.tail = node;
            return true;
        }
        let temp = this.head;
        while (temp.next != null) {
            temp = temp.next;
        }
        this.len++;
        temp.next = node;
        this.tail = node;
        return true;
    }
    size() {
        return this.len;
    }
    pop() {
        if (this.len <= 0) return null;
        if (this.len === 1) {
            this.len--;
            this.head = null;
            this.tail = null;
            return true;
        }
        let temp = this.head;
        while (temp.next.next != null) {
            temp = temp.next;
        }
        temp.next = null;
        this.len--;
        this.tail = temp;
        return true;
    }
    peek() {
        return this.tail.val;
    }
}

let stack = new Stack();
stack.push(100);
stack.push(200);
stack.push(300);
console.log(stack.peek());
